proposals/namespace_editing/README.md

```diff
@@ -0,0 +1,480 @@
+
+Note that delete, rename, and reparent operations will only be treated as such when performed through the official UsdNamespaceEditor API. In other words, we will not try to figure out from the standard Sdf change notifications if some set of layer changes performed through other API could be equivalent to a delete, rename, or reparent when populating the OjbectsChanged notice.
+
+#### Batch Editing
+
+Similar to how we have SdfBatchNamespaceEdit to describe a series of one or more Sdf namespace edits that we can then possibly apply more efficiently than we would be able to if applied one at a time, we will provide a USD namespace editing equivalent structure, such as UsdBatchNamespaceEdit. This batch edit structure will allow all of the same edit operations provided by UsdNamespaceEditor to be added in sequence as edits to the batch. The UsdNamespaceEditor class will have the functions ApplyEdits(batchNamespaceEdit) and CanApplyEdits(batchNamespaceEdit) that will perform (and determine if can be performed) the effect of the sequence of edits in the batch. Like with SdfBatchNamespaceEdit, the edits will be processed before they are applied to determine the most efficient way to perform the series of edits.
+
+Another option for batch editing would be to implement a BatchEditingBlock RAII structure that, when opened, would cause all edits performed via UsdNamespaceEditor to be queued up until the block is closed/deleted at which point the edits would be processed and performed. This has the advantage that we wouldn’t need an entire structure that parallels the same API as UsdNamespaceEditor just for batch editing. There's a potential downside in that we would have to process the effects of the queued edits to answer any of the “can perform edit” queries made during an open batch, but this may end up not being a big enough issue to avoid going this route. Another potential issue would result if the user wanted to change the authoring settings or edit target in the middle of an open batch. We could work around this by disallowing the changing of authoring settings or edit targets when a batch is open or by just using the state of the settings when the batch is closed.
```

Comment:
> We could work around this by disallowing the changing of authoring settings or edit targets when a batch is open or by just using the state of the settings when the batch is closed

Could we not also just have each UsdNamespaceEditor method snap and store the current UsdEditTarget at the time it is called, rather than storing just one to associate with the whole Editor/RAII-batcher ?

Reply:
That is a potential option, but I'm also concerned about the potential complexity in processing the effects of multiple edits spanning across many layers stacks that could have namespace dependencies on each other (and not to mention the extra complexity of properly testing it). From an implementation standpoint, I would definitely start with only allowing batch edits to involve a single edit target before trying to expand to multiple edit targets.